src/libraries/PCPricing.sol

```diff
@@ -0,0 +1,303 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+/// @title Preferred Currency Pricing (PCPricing) Library
+/// @author Permissionless Games & ChatGpt
+/// @notice This library enables dynamic price adjustments based on preferred currency usage.
+/// @dev When the anchor currency is used, the price of all other tracked currencies can be decreased.
+///      A universal adjustment factor applies to all non-anchor currencies, promoting dynamic pricing models.
```

Comment:
Explain the update rule in more mathematical detail in the comments or in other documentation.